backend/functions/caseassignments/case.assignment.function.test.ts

```diff
@@ -0,0 +1,187 @@
+import httpTrigger from './case.assignment.function';
+import { applicationContextCreator } from '../lib/adapters/utils/application-context-creator';
+import { CaseAssignmentController } from '../lib/adapters/controllers/case.assignment.controller';
+import * as httpModule from '../lib/adapters/utils/http';
+const context = require('azure-function-context-mock');
+
+const appContext = applicationContextCreator(context);
+describe('Case Assignment Function Tests', () => {
+  test('Return the function response with the assignment Id created for the new case assignment', async () => {
+    const request = {
+      query: {},
+      body: {
+        caseId: '6789',
+        attorneyIdList: ['9082'],
+        role: 'TrialAttorney',
+      },
+    };
+
+    const expectedResponse = {
+      assignmentIdList: [1],
+      success: true,
+      message: 'Trial attorney assignments created.',
+      resultCount: 1,
+    };
+    await httpTrigger(appContext, request);
+    expect(appContext.res.body).toEqual(expectedResponse);
+  });
+
+  test('returns response with multiple assignment Ids , when requested to create assignments for multiple trial attorneys on a case', async () => {
+    const request = {
+      query: {},
+      body: {
+        caseId: '6789',
+        attorneyIdList: ['2082', '2083'],
+        role: 'TrialAttorney',
+      },
+    };
+
+    const expectedResponse = {
+      assignmentIdList: [1, 2],
+      success: true,
+      message: 'Trial attorney assignments created.',
+      resultCount: 2,
+    };
+
+    await httpTrigger(appContext, request);
+    expect(appContext.res.body).toEqual(expectedResponse);
+  });
+  test('handle any duplicate attorney Ids passed in the request, not create duplicate assignments', async () => {
+    const request = {
+      query: {},
+      body: {
+        caseId: '6789',
+        attorneyIdList: ['2082', '2082'],
+        role: 'TrialAttorney',
+      },
+    };
+
+    const expectedResponse = {
+      assignmentIdList: [1],
+      success: true,
+      message: 'Trial attorney assignments created.',
+      resultCount: 1,
+    };
+
+    await httpTrigger(appContext, request);
+    expect(appContext.res.body).toEqual(expectedResponse);
+  });
+
+  test('returns bad request 400 when a caseId is not passed in the request', async () => {
+    const request = {
+      query: {},
+      body: {
+        caseId: '',
+        attorneyIdList: ['2082', '2083'],
+        role: 'TrialAttorney',
+      },
+    };
+    const expectedResponse = {
+      error: 'Required parameter caseId is absent.',
+    };
+
+    await httpTrigger(appContext, request);
+    expect(appContext.res.body).toEqual(expectedResponse);
+    expect(appContext.res.statusCode).toBe(400);
+    expect(appContext.res.body);
```

Comment:
I don't believe we need this (or the similar lines in a couple of the other tests).

Reply:
Resolving based on our discussion to use the status code to distinguish between bad request vs internal server error. Hence keeping these tests.

Reply:
What is the expect supposed to be validating?
```
expect(appContext.res.body);
```

There's no condition.